### PR TITLE
Add Airbyte CDK module scaffolding

### PR DESCRIPTION
**Pull request by sherifnada**
## Problem
tl;dr is idiomatic package management in python

Our story for consuming packages is wonky:
* we have 3 “base” packages: airbyte-protocol generated models, base-python, base-singer
* If I want to consume these packages, we do this in two different ways depending on the environment: locally or in Docker. Locally, I can do an editable install if I’m working in the monorepo (this is the -e ../../bases in requirements.txt). This works completely fine with the minor inconvenience of having requirements.txt and setup.py. I also have to work in the monorepo unless I want to do pip install base-python @ <link_to_specific_github_commit> . Fine — no big deal.
* To publish a docker image, I can’t really do that because requirements.txt is referencing modules outside the build context. So either I lift the build context to be all of airbyte-integrations/  buy more RAM and change all paths in my Dockerfile to be airbyte-integrations/connectors/my_connector etc.. accordingly. Or I find a way to get this dependency into my Docker image somehow.
* We chose the latter. Our implementation was to create three docker images corresponding to the 3 python packages. The relationship is hierarchical: base-singer inherits base-python which inherits airbyte-protocol. Each of these images globally installs the relevant python package. To use this in my connector module, I inherit the highest Docker image which fulfills my need (if you need base-python and airbyte_protocol, inherit base-python image etc..). Then I just specify the depenedency in setup.py. Because it’s already installed on the image, no further action is needed from Pip.

initially this setup had the problem that we didn’t publish these docker images, so a contributor had to run gradle to build these images. Then we said no more gradle, so we’ll version & publish the docker images. All this has led to a lot of problems:

Some problems with this:
1. all these packages are effectively different facets of the Airbyte Python CDK. I can’t think of a compelling reason why they should be 3 different packages.
2. we should not have two separate ways of installing dependencies for local and Docker
3. Consuming these packages currently forces me to use a specific python version. Instead the package should say which python version it has been tested with and give me freedom to choose.
4. in order to version manage the deps, I have to change the versions of the docker images I’m using, not setup.py. Plus, I cannot iterate locally within the monorepo on a previous version of the package - I have to use the newest one even if no breaking changes were made to the Airbyte protocol (but potentially to the python CDK)
5. (small concern) The Docker images force me to use an opaque entrypoint defined somewhere in the parent docker hierarchy.

This is all super complicated for no good reason. We are just publishing a python CDK for people to build connectors. So let’s just call it that, put it into one package, and use the idiomatic python workflow.

Hence the solution I’m proposing which is: merge everything into one python package and publish it to pypi 

## Goal End State 
1. Introduce the top level `airbyte-cdk` directory. This will contain the CDK in different languages e.g `airbyte-cdk/python, airbyte-cdk/java` etc.. Having all CDKs at the top level of the repo for ease of discoverability. This will require changing our intellij Python project setup but I don't think that should really be an issue other than the one-off inconvenience. 
2. `airbyte-cdk/python` will contain the merged package consisting of `airbyte-protocol`, `base-singer`, and `base-python`. This is published normally to Pypi. 

## This PR achieves
1. Introduce the top level `airbyte-cdk` directory. This will contain the CDK in different languages e.g `airbyte-cdk/python, airbyte-cdk/java` etc..
2. Copy paste airbyte_protocol into this module for now. This package is not used for anything at the moment. I'm merging to master to give Davin a base to work off of for publishing to pypi. 

3.  Next step is to merge base-python/base-singer into this and clean up the module.

### CODE REVIEW

**Review by davinchia — APPROVED** [review]

**Comment by davinchia** [comment]
Sherif and I spoke about this. I'm on board. To add to the points Sherif wrote out:
1) When setting this up in Intellij, I currently have to add `base_python` and `airbyte_protocol` as module dependencies in order for Python to find the modules `requirements.txt` states. Publishing to PyPi solves this.
2) Faster builds! Since this is no longer locally compiled for every source, and can be downloaded from PyPi, we should save unnecessary build time. Faster build!!!

**Review by jrhizor — APPROVED** [review]
Thank you.
